utils/remote.cc

```diff
@@ -1974,8 +1984,128 @@ static void printGroups(tr_variant* top)
+        bool negate = false;
+        std::string_view arg;
+
+        if (filter[pos] == '~')
+        {
+            ++pos;
+            negate = true;
+        }
+        if (strlen(filter) > pos + 1 && filter[pos + 1] == ':')
+        {
+            arg = filter + pos + 2;
+        }
+
+        for (int i = 0, n = tr_variantListSize(list); i < n; ++i)
+        {
+            tr_variant* d = tr_variantListChild(list, i);
+            int64_t torId;
+            if (!tr_variantDictFindInt(d, TR_KEY_id, &torId) || torId < 0)
+            {
+                continue;
+            }
+            /*
+            if (tr_variantDictFindInt(d, TR_KEY_eta, &eta) && 
+                tr_variantDictFindInt(d, TR_KEY_leftUntilDone, &leftUntilDone) &&
+                tr_variantDictFindStrView(d, TR_KEY_name, &name) && tr_variantDictFindInt(d, TR_KEY_rateDownload, &down) &&
+                tr_variantDictFindInt(d, TR_KEY_rateUpload, &up) &&
+                tr_variantDictFindInt(d, TR_KEY_sizeWhenDone, &sizeWhenDone) &&
+                && tr_variantDictFindReal(d, TR_KEY_uploadRatio, &ratio))
+                */
+            bool include = negate;
+            std::string status = getStatusString(d);
+            std::string_view name;
+            switch (filter[pos])
+            {
+            case 'i': // Status = Idle
+                if (status == "Idle")
+                {
+                    include = !include;
+                }
+                break;
+            case 'd': // Downloading (Status is Downloading or Up&Down)
+                if (status.find("Down") != std::string::npos)
+                {
+                    include = !include;
+                }
+                break;
+            case 'u': // Uploading (Status is Uploading, Up&Down or Seeding
+                if ((status.find("Up") != std::string::npos) || (status == "Seeding"))
+                {
+                    include = !include;
+                }
+                break;
+            case 'l': // label
+                tr_variant* l;
+                if (tr_variantDictFindList(d, TR_KEY_labels, &l))
+                {
+                    size_t child_pos = 0;
+                    tr_variant const* child;
+                    std::string_view sv;
+                    while ((child = tr_variantListChild(l, child_pos++)))
+                    {
+                        if (tr_variantGetStrView(child, &sv))
+                        {
+                            if (arg == sv)
+                            {
+                                include = !include;
+                                break;
+                            }
+                        }
+                    }
+                }
+                break;
+            case 'n': // Torrent name substring
+                if (!tr_variantDictFindStrView(d, TR_KEY_name, &name))
+                {
+                    continue;
+                }
+                if (name.find(arg) != std::string::npos)
```

Comment:
scope of `name` can be narrowed

```suggestion
                if (std::string_view name; !tr_variantDictFindStrView(d, TR_KEY_name, &name))
                {
                    continue;
                }
                else if (name.find(arg) != std::string::npos)
```